### ISSUE TITLE
Rules evaluate also deleted resources.

### ISSUE THREAD

**Issue by Tmorinaga** [deep-security/aws-config#4]
When I tried ds-IsInstanceProtectedBy rule, I found it evaluates terminated instances. 
I don't need to know deleted resources information.
AWS Config Rules pass a parameter `event['eventLeftScope']`, a flag whether resources have been deleted or not, to lambda. I think this parameter is key.

**Comment by marknca** [comment]
@Tmorinaga you're correct. I've added code in a new branch to evaluate the scope of the resources and stop evaluation when it encounters a deleted one.

Can you please test out [https://github.com/deep-security/aws-config/tree/out-of-scope](https://github.com/deep-security/aws-config/tree/out-of-scope) and let me know your results? Thanks!
